android/src/main/kotlin/com/rudderstack/sdk/kotlin/android/plugins/devicemode/eventprocessing/EventFilteringPlugin.kt

```diff
@@ -0,0 +1,128 @@
+package com.rudderstack.sdk.kotlin.android.plugins.devicemode.eventprocessing
+
+import com.rudderstack.sdk.kotlin.android.utils.findDestination
+import com.rudderstack.sdk.kotlin.android.utils.getString
+import com.rudderstack.sdk.kotlin.core.Analytics
+import com.rudderstack.sdk.kotlin.core.internals.logger.LoggerAnalytics
+import com.rudderstack.sdk.kotlin.core.internals.models.Event
+import com.rudderstack.sdk.kotlin.core.internals.models.TrackEvent
+import com.rudderstack.sdk.kotlin.core.internals.plugins.Plugin
+import com.rudderstack.sdk.kotlin.core.internals.utils.LenientJson
+import com.rudderstack.sdk.kotlin.core.internals.utils.empty
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.launch
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonObject
+import java.util.concurrent.CopyOnWriteArrayList
+
+private const val WHITE_LIST_EVENTS = "whitelistedEvents"
+private const val BLACK_LIST_EVENTS = "blacklistedEvents"
+private const val EVENT_FILTERING_OPTION = "eventFilteringOption"
+
+/**
+ * A plugin to filter events based on the event filtering option provided in the destination config.
+ *
+ * This plugin filters the events based on the event filtering option provided in the destination config.
+ * The plugin supports two types of event filtering options based on the dashboard configuration:
+ * 1. Whitelist events: Only the events present in the whitelist will be allowed.
+ * 2. Blacklist events: All the events except the ones present in the blacklist will be allowed.
+ */
+internal class EventFilteringPlugin(private val key: String) : Plugin {
+
+    override val pluginType: Plugin.PluginType = Plugin.PluginType.PreProcess
+
+    override lateinit var analytics: Analytics
+
+    @Volatile
+    private var filteringOption = String.empty()
+    private val filteringList = CopyOnWriteArrayList<String>()
+
+    override fun setup(analytics: Analytics) {
+        super.setup(analytics)
+        listenForConfigChanges()
+    }
+
+    override suspend fun intercept(event: Event): Event? {
+        if (event !is TrackEvent) {
+            return event
+        }
+
+        val eventName = event.event.trim()
+        return when {
+            shouldDropEvent(eventName) -> {
+                LoggerAnalytics.debug("EventFilteringPlugin: Dropped event '$eventName' for destination: $key")
+                null
+            }
+            else -> event
+        }
```

Comment:
NIT: We can improve the code.
```suggestion
        return when(shouldDropEvent(eventName)) {
            true -> {
                LoggerAnalytics.debug("EventFilteringPlugin: Dropped event '$eventName' for destination: $key")
                null
            }
            false -> event
        }
```